src/main/java/appeng/core/sync/packets/PacketInventoryAction.java

```diff
@@ -107,66 +78,106 @@ public PacketInventoryAction( final InventoryAction action, final int slot, fina
-
-					if( sender.openContainer instanceof ContainerCraftAmount )
+					ClientHelper.proxy.getPlayers().get( 0 ).inventory.setItemStack( null );
+				}
+				else
+				{
+					ClientHelper.proxy.getPlayers().get( 0 ).inventory.setItemStack( message.slotItem.getItemStack() );
+				}
+			}
+		}
+		else
+		{
+			final EntityPlayerMP sender = (EntityPlayerMP) ctx.getServerHandler().playerEntity;
+			if( sender.openContainer instanceof AEBaseContainer )
+			{
+				final AEBaseContainer baseContainer = (AEBaseContainer) sender.openContainer;
+				if( message.action == InventoryAction.AUTO_CRAFT )
+				{
+					final ContainerOpenContext context = baseContainer.openContext;
+					if( context != null )
 					{
-						final ContainerCraftAmount cca = (ContainerCraftAmount) sender.openContainer;
+						final TileEntity te = context.getTile();
+						Platform.openGUI( sender, te, baseContainer.openContext.side, GuiBridge.GUI_CRAFTING_AMOUNT );
 
-						if( baseContainer.getTargetStack() != null )
+						if( sender.openContainer instanceof ContainerCraftAmount )
 						{
-							cca.craftingItem.putStack( baseContainer.getTargetStack().getItemStack() );
-							cca.whatToMake = baseContainer.getTargetStack();
-						}
+							final ContainerCraftAmount cca = (ContainerCraftAmount) sender.openContainer;
+
+							if( baseContainer.getTargetStack() != null )
+							{
+								cca.craftingItem.putStack( baseContainer.getTargetStack().getItemStack() );
+								cca.whatToMake = baseContainer.getTargetStack();
+							}
 
-						cca.detectAndSendChanges();
+							cca.detectAndSendChanges();
+						}
 					}
 				}
+				else
+				{
+					baseContainer.doAction( sender, message.action, message.slot, message.id );
+				}
 			}
-			else
+		}
+
+		return null;
+	}
+
+	@Override
+	public void fromBytes( final ByteBuf buf )
+	{
+		this.action = InventoryAction.values()[buf.readInt()];
+		this.slot = buf.readInt();
+		this.id = buf.readLong();
+		final boolean hasItem = buf.readBoolean();
+		if( hasItem )
+		{
+			try
+			{
+				this.slotItem = AEItemStack.loadItemStackFromPacket( buf );
+			}
+			catch( final IOException e )
 			{
-				baseContainer.doAction( sender, this.action, this.slot, this.id );
 			}
 		}
+		else
+		{
+			this.slotItem = null;
+		}
 	}
 
 	@Override
-	public void clientPacketData( final INetworkInfo network, final AppEngPacket packet, final EntityPlayer player )
+	public void toBytes( final ByteBuf buf )
 	{
-		if( this.action == InventoryAction.UPDATE_HAND )
+		buf.writeInt( this.action.ordinal() );
+		buf.writeInt( this.slot );
+		buf.writeLong( this.id );
+
+		if( this.slotItem == null )
+		{
+			buf.writeBoolean( false );
+		}
+		else
 		{
-			if( this.slotItem == null )
+			buf.writeBoolean( true );
+			try
 			{
-				ClientHelper.proxy.getPlayers().get( 0 ).inventory.setItemStack( null );
+				this.slotItem.writeToPacket( buf );
 			}
-			else
+			catch( final IOException e )
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)